x-pack/plugins/ml/public/application/model_management/expanded_row.tsx

```diff
@@ -169,13 +169,40 @@ export const ExpandedRow: FC<ExpandedRowProps> = ({ item }) => {
     license_level,
   ]);
 
-  const deploymentStatItems: AllocatedModel[] = useMemo<AllocatedModel[]>(() => {
+  const deploymentStatItems = useMemo<AllocatedModel[]>(() => {
     const deploymentStats = stats.deployment_stats;
     const modelSizeStats = stats.model_size_stats;
 
     if (!deploymentStats || !modelSizeStats) return [];
 
-    const items: AllocatedModel[] = deploymentStats.flatMap((perDeploymentStat) => {
+    return deploymentStats.flatMap((perDeploymentStat) => {
+      // A deployment can be in a starting state and not allocated to any node yet.
+      if (perDeploymentStat.nodes.length < 1) {
+        return [
+          {
+            key: `${perDeploymentStat.deployment_id}_no_node`,
+            ...perDeploymentStat,
+            ...modelSizeStats,
+            node: {
```

Comment:
One issue with having this node info with hard coded 0s is that when new stats are added, or renaming takes place on the back end, we're going to have keep this in sync. Which I think will be hard to do.
Would it be possible to show a "node info not available" message instead, or something like that?

Reply:
It is type-safe, useMemo has the `AllocatedModel` type, hence it takes care of the return type, and IDEs will rename properties when the underlying interface is changed.

Reply:
I also have a [PR in progress ](https://github.com/elastic/kibana/pull/200977) that updates the types and optimizes stats fetching, so the structure will differ. 
I kept these changes to a minimum to backport to 8.16.2 as well.